rfcs/0000-tsdb-dimensions.md

```diff
@@ -0,0 +1,240 @@
+  type: group
+  fields:
+    - name: hostname
+      level: core
+      type: keyword
+      short: Hostname of the host.
+      dimension: true
+      description: >
+        Hostname of the host.
+
+        It normally contains what the `hostname` command returns on the host machine.
+
+```
+Changes to host mapping
+
+```yaml
+---
+- name: agent
+  title: Agent
+  group: 2
+  short: Fields about the monitoring agent.
+  description: >
+    The agent fields contain the data about the software entity, if any, that collects, detects, or observes events on a host, or takes measurements on a host.
+
+    Examples include Beats. Agents may also run on observers. ECS agent.* fields shall be populated with details of the agent running on the host or observer where the event happened or the measurement was taken.
+  footnote: >
+    Examples: In the case of Beats for logs, the agent.name is filebeat. For APM, it is the
+    agent running in the app/service. The agent information does not change if
+    data is sent through queuing systems like Kafka, Redis, or processing systems
+    such as Logstash or APM Server.
+  type: group
+  fields:
+
+    - name: id
+      level: core
+      type: keyword
+      short: Unique identifier of this agent.
+      description: >
+        Unique identifier of this agent (if one exists).
+
+        Example: For Beats this would be beat.id.
+      example: 8a4f500d
+      dimension: true
+```
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+<!--
+Stage 2: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+Integration package development is the key beneficiary of this change. The fields of the document that are received from an integration receives a field mapping. If and when TSDB benefits are to be utilised, along with the field mapping with a metric type, at least one of the fields  must receive `dimension: true` annotation. 
```

Comment:
I expect this eventually to become the default mappings for any `metrics-*-*` data stream so it will benefit all metrics cases: https://github.com/elastic/elasticsearch/issues/85692